.ibm/pipelines/cluster/osd-gcp/create-osd.sh

```diff
@@ -59,22 +64,19 @@ echo "    api_url: $(ocm describe cluster $CLUSTER_ID --json | jq -rc '.api.url'
 echo "    $KUBEADMIN_USER: $KUBEADMIN_PASSWORD" >> $CLUSTER_CONFIG
 echo "    cli: oc login $(ocm describe cluster $CLUSTER_ID --json | jq -rc '.api.url') --username $KUBEADMIN_USER --password $KUBEADMIN_PASSWORD" >> $CLUSTER_CONFIG
 
-echo
-cat $CLUSTER_CONFIG
-echo
-
 export KUBECONFIG=$WORKSPACE/kubeconfig
 rm -rvf $KUBECONFIG
-retries=50
-until [[ $retries == 0 ]]; do
-    echo "Attempting to login to get kubeconfig - $retries attempts remaining..."
-    oc login $(ocm describe cluster $CLUSTER_ID --json | jq -rc '.api.url') --username $KUBEADMIN_USER --password $KUBEADMIN_PASSWORD >/dev/null 2>&1 && break
+CLUSTER_API_URL=$(ocm describe cluster $CLUSTER_ID --json | jq -rc '.api.url')
```

Comment:
Validation of CLUSTER_API_URL: Ensures the API URL is correctly retrieved before attempting to log in. If the retrieval fails, the script exits early with an error message.
For example:
CLUSTER_API_URL=$(ocm describe cluster "$CLUSTER_ID" --json | jq -rc '.api.url')

if [[ -z "$CLUSTER_API_URL" ]]; then
    echo "Error: Failed to retrieve CLUSTER_API_URL. Ensure CLUSTER_ID is correct and OCM is configured."
    exit 1
fi